src/azul/plugins/__init__.py

```diff
@@ -191,25 +194,44 @@ def sources(self) -> AbstractSet[SOURCE_SPEC]:
         """
         raise NotImplementedError
 
-    def resolve_source(self, *, spec: str, id: Optional[str] = None) -> SOURCE_REF:
+    @abstractmethod
+    def list_sources(self,
+                     authentication: Optional[Authentication]
+                     ) -> Iterable[SOURCE_REF]:
+        """
+        The sources the plugin is configured to read metadata from.
+        Retrieving this information may require a round-trp to the underlying
+        repository. Implementations should raise PermissionError if the provided
+        authentication is insufficient to access the repository.
+        """
+        raise NotImplementedError
+
+    def resolve_source(self,
+                       *,
+                       spec: str,
+                       id: Optional[str] = None,
+                       verify_id: bool = True) -> SOURCE_REF:
```

Comment:
With the new argument set to False, this alters the behavior of the method beyond recognition. It's name suggests that it resolves something, and the ID lookup is that resolution. See REVIEW commit.